torch/package/_package_utils.py

```diff
@@ -0,0 +1,60 @@
+import re
+from typing import List
+
+
+def _print_file_structure(
+    filename: str, records: List[str], skip_storage: bool = True, path_includes=".*"
+):
+    space = "    "
+    branch = "│   "
+    tee = "├── "
+    last = "└── "
+    start = "─── "
+
+    class Folder:
+        def __init__(self, name: str):
+            self.name = name
+            self.sub_folders = {}
+            self.sub_files = []
+
+        # builds path of folders if not yet built, returns last folder
+        def get_folder(self, folders: List[str]):
+            if len(folders) == 0:
+                return self
+            folder_name = folders[0]
+            if folder_name not in self.sub_folders:
+                self.sub_folders[folder_name] = Folder(folder_name)
+            return self.sub_folders[folder_name].get_folder(folders[1:])
+
+        def add_file(self, file_path):
+            *folders, file = file_path.split("/")
+            folder = self.get_folder(folders)
+            folder.sub_files.append(file)
+
+        def stringify_tree(
+            self, str_list: List[str], extention: str = "", folder_ptr: str = start
+        ):
+            self.sub_files.sort()
+            str_list.append(f"{extention}{folder_ptr}{self.name}\n")
+            if folder_ptr == tee:
+                extention = extention + branch
+            else:
+                extention = extention + space
+            for index, key in enumerate(sorted(self.sub_folders)):
+                if (index == len(self.sub_folders) - 1) and len(self.sub_files) == 0:
+                    self.sub_folders[key].stringify_tree(str_list, extention, last)
+                else:
+                    self.sub_folders[key].stringify_tree(str_list, extention, tee)
+            for index, file in enumerate(self.sub_files):
+                if skip_storage and ".storage" in file:
+                    continue
+                pointer = last if (index == len(self.sub_files) - 1) else tee
+                str_list.append(f"{extention}{pointer}{file}\n")
+
+    top_folder = Folder(filename)
+    for file in records:
+        if re.search(path_includes, file):
+            top_folder.add_file(file)
+    str_list: List[str] = []
+    top_folder.stringify_tree(str_list)
+    print("".join(str_list))
```

Comment:
I wasn't sure if either printing out or returning the file structure as a string would be a better user experience. This PR goes with printing out the file structure, but I can change it

Reply:
Probably return the string. It's easy to print a string, it is hard to make a string out of something printed to stdout.

Reply:
Cool, changed to the string